### DISCUSSION TITLE
Why an API request is made in SSG with revalidate: false?

### DISCUSSION THREAD

**Discussion by wfelipe99** [urql-graphql/urql#1890] [Q&A] [answered]
Everything is working fine, I'm just trying to understand things.

I have a component using SSG and `revalidate: false`. I build the project and the JSON/HTML is generated correctly. But when I serve the project, urql always make a request and get fresh data. But, let's suppose that this data won't ever be changed. So, what's the point to make a request?

Then, I have disabled javascript and it worked like I've expected: when I open the page, the data that is in the JSON file is served and no API request is made.

What I would like to know is:

1 - Is this behavior expected (to always make a request even if `revalidate: false`)?
2 - If yes, is there a way I can disable it to not make any request and serve the data that is in the JSON file that was generated in the build process? I'm asking this because the way urql is working now, if I throttle the network for slow 3G connection I first get `if (fetching) return <span>Getting data...</span>` and then I see the page. And this isn't the way I expect SSG to work, because its data is available since the build process.

<details><summary>Here's the code if it helps understand better</summary>
<p>

```typescript
import { useRouter } from 'next/router'
import Head from 'next/head'
import { gql, useQuery } from 'urql'
import { withUrqlClient } from 'next-urql'
import { client, ssrCache } from 'utils/urql'
import db from 'utils/prisma'

const TOPIC_QUERY = gql`
  query GetTopic($topicId: Int!) {
    topic(input: { id: $topicId }) {
      id
      name
      description
    }
  }
`

function Topic() {
  const router = useRouter()
  const { id } = router.query
  const [result] = useQuery({ query: TOPIC_QUERY, variables: { topicId: Number(id) } })

  const { fetching } = result

  if (fetching) return <p>Getting data...</p>

  const {
    data: { topic },
  } = result

  return (
    <>
      <Head>
        <title>{topic.name}</title>
      </Head>
      <p>
        {topic.name} - {topic.description}
      </p>
    </>
  )
}

export async function getStaticProps({ params }) {
  await client.query(TOPIC_QUERY, { topicId: Number(params.id) }).toPromise()

  return {
    props: { urqlState: ssrCache.extractData() },
    revalidate: false,
  }
}

export async function getStaticPaths() {
  const topics = await db.topic.findMany({ select: { id: true } })

  const paths = topics.map((topic) => ({ params: { id: `${topic.id}` } }))

  return { paths, fallback: false }
}

export default withUrqlClient(
  (ssr) => ({
    url: `${process.env.NEXT_PUBLIC_API_URL}`,
  }),
  { ssr: false }
)(Topic)

```

</p>
</details>

Thank you in advance.

**Comment by JoviDeCroock** [comment, marked as answer]
Hey

i've just tried this myself and am not running into the issue 😅 https://codesandbox.io/s/inspiring-rubin-52wrc?file=/pages/index.js the issue could reside in the difference of our `getStaticProps` function or the `exchanges` you're using in your client

**Reply by wfelipe99** [reply]
Hey, thank you for answering.

Finally I got it to work right. You're correct. The problem was my code. I was using the same exchanges, but I was wrapping `_app.tsx` with urql's provider. Do you have a succinct answer why this was happening? I will never use the provider in any case in a Next project?

**Reply by JoviDeCroock** [reply]
If that one is without ssr: true you'll apply everything twice and use prepass/clear urqlState

**Reply by wfelipe99** [reply]
Got it. Thank you again!
